rio_tiler/io/base.py

```diff
@@ -99,3 +97,261 @@ def preview(self, **kwargs: Any) -> Tuple[numpy.ndarray, numpy.ndarray]:
+        """Parse rio-tiler band math expression."""
+        _re = re.compile("|".join(sorted(self.assets, reverse=True)))
+        return list(set(re.findall(_re, expression)))
+
+    def info(
+        self, assets: Union[Sequence[str], str] = None, *args, **kwargs: Any
+    ) -> Dict:
+        """Return metadata from multiple assets"""
+        if not assets:
+            raise MissingAssets("Missing 'assets' option")
+
+        if isinstance(assets, str):
+            assets = (assets,)
+
+        def _reader(asset: str, **kwargs: Any) -> Dict:
+            url = self._get_asset_url(asset)
+            with self.reader(url, **self.reader_options) as cog:  # type: ignore
+                return cog.info()
+
+        return multi_values(assets, _reader, *args, **kwargs)
+
+    def stats(
+        self,
+        pmin: float = 2.0,
+        pmax: float = 98.0,
+        hist_options: Optional[Dict] = None,
+        assets: Union[Sequence[str], str] = None,
+        **kwargs: Any,
+    ) -> Dict:
+        """Return array statistics from multiple assets"""
+        if not assets:
+            raise MissingAssets("Missing 'assets' option")
+
+        if isinstance(assets, str):
+            assets = (assets,)
+
+        def _reader(asset: str, *args, **kwargs) -> Dict:
+            url = self._get_asset_url(asset)
+            with self.reader(url, **self.reader_options) as cog:  # type: ignore
+                return cog.stats(*args, **kwargs)
+
+        return multi_values(
+            assets, _reader, pmin, pmax, hist_options=hist_options, **kwargs
+        )
+
+    def metadata(
+        self,
+        pmin: float = 2.0,
+        pmax: float = 98.0,
+        assets: Union[Sequence[str], str] = None,
+        **kwargs: Any,
+    ) -> Dict:
+        """Return metadata from multiple assets"""
+        if not assets:
+            raise MissingAssets("Missing 'assets' option")
+
+        if isinstance(assets, str):
+            assets = (assets,)
+
+        def _reader(asset: str, *args, **kwargs) -> Dict:
+            url = self._get_asset_url(asset)
+            with self.reader(url, **self.reader_options) as cog:  # type: ignore
+                return cog.metadata(*args, **kwargs)
+
+        return multi_values(assets, _reader, pmin, pmax, **kwargs)
+
+    def tile(
+        self,
+        tile_x: int,
+        tile_y: int,
+        tile_z: int,
+        assets: Union[Sequence[str], str] = None,
+        expression: Optional[str] = "",
+        asset_expression: Optional[
+            str
+        ] = "",  # Expression for each asset based on index names
+        **kwargs: Any,
+    ) -> Tuple[numpy.ndarray, numpy.ndarray]:
+        """Read a Mercator Map tile multiple assets."""
+        if not tile_exists(self.bounds, tile_z, tile_x, tile_y):
+            raise TileOutsideBounds(
+                "Tile {}/{}/{} is outside image bounds".format(tile_z, tile_x, tile_y)
+            )
```

Comment:
I think we should remove this. Checking if the tile is outside the bounds here lets us raise before doing more file reading BUT it makes the class not really extendable for `rio-tiler-crs`